### PR TITLE
Subscription alterationType feature

### PR DESCRIPTION
**Pull request by fgalan**
Issue #1494

Pending:

- [x] CNR
- [x] Documentation

### CODE REVIEW

File: test/functionalTest/cases/1494_subscription_operations/sub_operation_entity_change.test [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    ],
+    "condition": {
+      "operations": [ "entityChange" ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://127.0.0.1:'${LISTENER_PORT}'/notify"
+    },
+    "attrs": [ "entityStatus", "*" ]
```
Comment (fgalan) [inline test/functionalTest/cases/1494_subscription_operations/sub_operation_entity_change.test:60]: As alternantives to entityStatus, it has been suggested:

* triggeredBy
* notifTriggeredBy

Comment (fgalan) [reply test/functionalTest/cases/1494_subscription_operations/sub_operation_entity_change.test:62]: NTC in this PR. See https://github.com/telefonicaid/fiware-orion/issues/1494#issuecomment-1076235243

File: src/lib/cache/subCache.cpp [hunk 2]
```diff
@@ -735,34 +786,35 @@ void subCacheItemInsert(CachedSubscription* cSubP)
...
-  bool                               onlyChanged
+  const char*                             tenant,
+  const char*                             servicePath,
+  const ngsiv2::HttpInfo&                 httpInfo,
+  const ngsiv2::MqttInfo&                 mqttInfo,
+  const std::vector<ngsiv2::EntID>&       entIdVector,
+  const std::vector<std::string>&         attributes,
+  const std::vector<std::string>&         metadata,
+  const std::vector<std::string>&         conditionAttrs,
+  const std::vector<ngsiv2::SubAltType>&  altTypes,
```
Comment (fgalan) [inline src/lib/cache/subCache.cpp:797]: The only actual modification in this block (other modified lines are just indent adjustment)

NTC (informative)

File: test/functionalTest/cases/1678_geosubscriptions/geosubscriptions.test [hunk 3]
```diff
@@ -42,7 +42,7 @@ accumulatorStart --pretty-print
 # 09. With subscription in place, create entity E3 inside area A (notif)
 # 10. Update subscription using a micro-area centered at E4
 # 11. With subscription in place, create entity E4 outside area A (notif)
-# 12. Dump acumulator (4 notifs: E1, E2, E1, E1, E3, E4)
+# 12. Dump acumulator (6 notifs: E1, E2, E1, E1, E3, E4)
```
Comment (fgalan) [inline test/functionalTest/cases/1678_geosubscriptions/geosubscriptions.test:45]: Small typo fix in this .test. Not related with the PR topic itself, but spotted while I was debugging cases.

NTC (informative)

File: test/functionalTest/cases/1494_subscription_alteration_types/sub_alteration_types_basic_crud.test [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+brokerStart CB 0-255
+
+--SHELL--
+
+#
+# 01. Create subscription with invalid alterationTypes, see error
+# 02. Create subscription with alterationTypes entityDelete and entityCreate
+# 03a. Get subscription and see alterationTypes entityDelete and entityCreate
+# 03b. Get subscription (as a list) and see alterationTypes entityDelete and entityCreate
+# 04. Update subscription with alterationTypes entityChange
```
[the remark this answers is not in the document]
Comment (fgalan) [reply test/functionalTest/cases/1494_subscription_alteration_types/sub_alteration_types_basic_crud.test:37]: Fixed in 65a9cc476. New steps 08 to 11.

**Review by mapedraza — APPROVED** [review]
LGTM

**Comment by fgalan** [comment]
@fisuda this PR does some modifications in .md files. It would be great if you could synchronize the Japanese translation, please. Thanks!

**Comment by fisuda** [comment]
I sent the PR #4089. Thanks.

**Pull request merged**